src/crl.rs

```diff
@@ -484,64 +486,76 @@ impl<'a> BorrowedRevokedCert<'a> {
+                // of a CRL revoked certificate's serial number is less than 20 octets in encoded form.
+                let serial_number = lenient_certificate_serial_number(der)
+                    .map_err(|_| Error::InvalidSerialNumber)?
+                    .as_slice_less_safe();
+
+                let revocation_date = Time::from_der(der)?;
+
+                let mut revoked_cert = BorrowedRevokedCert {
+                    serial_number,
+                    revocation_date,
+                    reason_code: None,
+                    invalidity_date: None,
+                };
+
+                // RFC 5280 §5.3:
+                //   Support for the CRL entry extensions defined in this specification is
+                //   optional for conforming CRL issuers and applications.  However, CRL
+                //   issuers SHOULD include reason codes (Section 5.3.1) and invalidity
+                //   dates (Section 5.3.2) whenever this information is available.
+                if der.at_end() {
+                    return Ok(revoked_cert);
+                }
 
-            // It would be convenient to use der::nested_of_mut here to unpack a SEQUENCE of one or
-            // more SEQUENCEs, however CAs have been mis-encoding the absence of extensions as an
-            // empty SEQUENCE so we must be tolerant of that.
-            let ext_seq = der::expect_tag(der, Tag::Sequence)?;
-            if ext_seq.is_empty() {
-                return Ok(revoked_cert);
-            }
+                // It would be convenient to use der::nested_of_mut here to unpack a SEQUENCE of one or
+                // more SEQUENCEs, however CAs have been mis-encoding the absence of extensions as an
+                // empty SEQUENCE so we must be tolerant of that.
+                let ext_seq = der::expect_tag(der, Tag::Sequence)?;
+                if ext_seq.is_empty() {
+                    return Ok(revoked_cert);
+                }
 
-            let mut reader = untrusted::Reader::new(ext_seq);
-            loop {
-                der::nested(&mut reader, Tag::Sequence, Error::BadDer, |ext_der| {
-                    // RFC 5280 §5.3:
-                    //   If a CRL contains a critical CRL entry extension that the application cannot
-                    //   process, then the application MUST NOT use that CRL to determine the
-                    //   status of any certificates.  However, applications may ignore
-                    //   unrecognized non-critical CRL entry extensions.
-                    revoked_cert.remember_extension(&Extension::from_der(ext_der)?)
-                })?;
-                if reader.at_end() {
-                    break;
+                let mut reader = untrusted::Reader::new(ext_seq);
+                loop {
+                    der::nested(
+                        &mut reader,
+                        Tag::Sequence,
+                        Error::TrailingData("BorrowedRevokedCert::inner"),
```

Comment:
maybe instead of `inner` -> `extensions`? Or `crlEntryExtensions` if you want to match 5280.

Reply:
Yup, used `BorrowedRevokedCert::crl_entry_extensions` here.